### ISSUE TITLE
When templater=dbt, AM04 - ambiguous.column_count is not reported as a violation

### ISSUE THREAD

**Issue by proetman-autogeneral** [sqlfluff/sqlfluff#6032]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When the templater = DBT, AM04 - ambiguous.column_count is not reported as a violation.
When sqlfluff is run with the -vvvvv option, the violation is raised in the log file, but not reported as an error.

### Expected Behaviour

The error AM04 should be reported as an error.

### Observed Behaviour

The error AM04 is NOT reported as an error.

### How to reproduce

command (run from the dbt project directory)
```
sqlfluff lint models\staging\claims\Home_Worklist\stg_frontline_home_worklist_testonly.sql
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
All Finished 📜 🎉!
```
stg_frontline_home_worklist_testonly.sql
```
{{  
    config
    (
        persist_docs={"relation": true, "columns": true}
    )
}}

select pahst.*
from {{ source('disc_proadm_raw','PAHST') }} pahst

```
Extract from log generated with -vvvvv option, please note the AM04 violation found log message
```
Entering linter phase main, loop 1/1

DEBUG      [AL05] Overlapping references found. Assuming bigquery semi-structured.
DEBUG      [AM04] Analyzing query: select pahst.*
from `data-lake-prod`.`disc_proadm_raw`.`PAHST` pahst
DEBUG      [AM04] Wildcard: pahst.* has target {wildcard_table}
DEBUG      [AM04] Query target `data-lake-prod-31d7`.`disc_proadm_raw`.`PAHST` is external. Generating warning.
INFO       [AM04] !! Violation Found: 'Query produces an unknown number of result columns.'
INFO       [CP01] _handle_segment: <KeywordSegment: ([L:  8, P:  1]) 'select'>, keyword
DEBUG      [CP01] Selected 'capitalisation_policy': 'consistent' from options ['upper', 'lower', 'capitalise']
...
...
== [models\staging\claims\Home_Worklist\stg_frontline_home_worklist_testonly.sql] PASS
==== summary ====
files:             1 violations:        0
clean files:       1 unclean files:     0
avg per file:   0.00 unclean rate:     0%
status:         PASS
All Finished 📜 🎉!
```

### Dialect

bigquery

### Version

(dbtenv) C:\source\python-gcp-dbt\projects\dai>sqlfluff --version
sqlfluff, version 3.1.0

(dbtenv) C:\source\python-gcp-dbt\projects\dai>python --version
Python 3.11.8

(dbtenv) C:\source\python-gcp-dbt\projects\dai>dbt --version
Core:
  - installed: 1.8.3
  - latest:    1.8.4 - Update available!

  Your version of dbt-core is out of date!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation

Plugins:
  - bigquery: 1.8.2 - Up to date!

Conda DBT pacakges
```
	Line  42: dbt-adapters              1.3.3              pyhd8ed1ab_0    conda-forge
	Line  43: dbt-bigquery              1.8.2              pyhd8ed1ab_0    conda-forge
	Line  44: dbt-common                1.6.0              pyhd8ed1ab_0    conda-forge
	Line  45: dbt-core                  1.8.3              pyhd8ed1ab_0    conda-forge
	Line  46: dbt-extractor             0.5.1           py311hc37eb10_1    conda-forge
	Line  47: dbt-semantic-interfaces   0.5.1              pyhd8ed1ab_0    conda-forge
	Line 204: sqlfluff-templater-dbt    3.1.0                    pypi_0    pypi
```

Note - we cannot upgrade to dbt-core version 1.8.4 due to incompatible with bigquery 1.8.2 (latest)

### Configuration

.sqlfluffignore
```
target/
# dbt <1.0.0
dbt_modules/
# dbt >=1.0.0
dbt_packages/
# macros/
```

.sqlfluff config:

```
[sqlfluff]
dialect = bigquery
templater = dbt
sql_file_exts = .sql,.dml,.ddl

max_line_length = 150

apply_dbt_builtins = True
unwrap_wrapped_queries = True

processes = -1

exclude_rules = 
	aliasing.table,
	aliasing.forbid,
	aliasing.self_alias.column,
	ambiguous.order_by,
	ambiguous.join,
    convention.not_equal,
    convention.coalesce,
    convention.select_trailing_comma,
    convention.count_rows,
    convention.terminator,
    convention.left_join,
    convention.blocked_words,
    layout.spacing,
    layout.cte_newline,
	layout.select_targets,
    references.from,
    references.consistent,
	references.special_chars,
    structure.column_order,
    structure.using,
    structure.join_condition_order

    
# If using the dbt templater, we recommend setting the project dir.
[sqlfluff:templater:dbt]
project_dir = ./
profiles_dir = ./
load_macros_from_path = ./macros


[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable, default='') %}item{% endmacro %}
dbt_is_incremental = {% macro is_incremental() %}True{% endmacro %}

[sqlfluff:indentation]
allow_implicit_indents = True
indented_joins = True

[sqlfluff:rules:aliasing.length]
min_alias_length = 3
max_alias_length = 60

# AM06 Inconsistent column references in GROUP BY/ORDER BY clauses
[sqlfluff:ambiguous.column_references]
group_by_and_order_by_style = consistent

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = consistent

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = consistent

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = consistent

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = consistent

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = consistent

# The default configuration for the not equal convention rule is "consistent"
# which will auto-detect the setting from the rest of the file. This
# is less desirable in a new project and you may find this (slightly
# more strict) setting more useful.
[sqlfluff:rules:convention.not_equal]
# Default to preferring the "c_style" (i.e. `!=`)
preferred_not_equal_style = c_style

[sqlfluff:rules:convention.quoted_literals]
preferred_quoted_literal_style = consistent

[sqlfluff:rules:convention.casting_style]
preferred_type_casting_style = consistent

# forbid subquery in both from and join statements
[sqlfluff:rules:structure.subquery]
forbid_subquery_in = both         
``` 

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
